modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-repo/src/test/java/org/nuxeo/apidoc/test/TestSecureXML.java

```diff
@@ -15,32 +15,174 @@
+    }
+
+    @Test
+    public void testSecureXMLWhitelistedAttribute() throws Exception {
+        assertEquals("<prop name=\"passwordField\">password</prop>",
+                SecureXMLHelper.secure("<prop name=\"passwordField\">password</prop>"));
+    }
+
+    @Test
+    public void testSecureXMLWhitelistedAttributeSpace() throws Exception {
+        assertEquals("<prop name=\"passwordField\" >password</prop>",
+                SecureXMLHelper.secure("<prop name=\"passwordField\" >password</prop>"));
+    }
+
+    @Test
+    public void testSecureXMLWhitelistedAttribute2() throws Exception {
+        assertEquals("<prop passwordField=\"password\">********</prop>",
+                SecureXMLHelper.secure("<prop passwordField=\"password\">foo</prop>"));
+    }
+
+    @Test
+    public void testSecureXMLWhitelistedAttribute3() throws Exception {
+        assertEquals("<prop passwordField=\"password\" foo=\"bar\"/>",
+                SecureXMLHelper.secure("<prop passwordField=\"password\" foo=\"bar\"/>"));
+    }
+
     @Test
     public void testSecureXML() throws Exception {
-        String xml = "foo <password>p1</password> " //
-                + " <myPassword>p2</myPassword>" //
-                + " <yo password=\"p3\" name=\"bla\">" //
-                + " <yo otherPassword=\"p4\" name=\"bla\">" //
-                + " <prop name=\"password\">p5</prop>" //
-                + " <prop name=\"realPassword\">p6</prop>";
-        String expected = "foo <password>********</password> " //
-                + " <myPassword>********</myPassword>" //
-                + " <yo password=\"********\" name=\"bla\">" //
-                + " <yo otherPassword=\"********\" name=\"bla\">" //
-                + " <prop name=\"password\">********</prop>" //
-                + " <prop name=\"realPassword\">********</prop>";
-        String res = DocumentationHelper.secureXML(xml);
-        assertEquals(expected, res);
+        String xml = "foo <password>p1</password>\n" //
+                + " <myPassword>p2</myPassword>\n" //
+                + " <myPassword >p2</myPassword>\n" //
+                + " <yo password=\"p3\" other=\"bla\">\n" //
+                + " <yo otherPassword=\"p4\" other=\"bla\">\n" //
+                + " <prop name=\"password\">p5</prop>\n" //
+                + " <prop name=\"realPassword\">p6</prop>\n" //
+                + " <prop name=\"realPassword\" >p6</prop>\n" //
+                + " <prop name=\"passwordNotWhitelisted\">p7</prop>\n" //
+                + " <prop name=\"passwordField\">ok</prop>\n" //
+                + " <secret>${nuxeo.jwt.secret}</secret>\n" //
+                + " <secretKey>${nuxeo.aws.secretKey}</secretKey>\n" //
+                + " <option name=\"apiKey\">${metrics.datadog.apiKey}</option>\n" //
+                + " <passwordField>password</passwordField>\n" //
+                + " <passwordHashAlgorithm>SSHA</passwordHashAlgorithm>";
```

Comment:
It seems to me there is a case that is not tested ?
`<password></password>`
should also be replaced with
`<password>********</password>`as you do have `[^<]*` in your regex.
An empty password wouldn't be hard to find anyway

Reply:
It's tested here: https://github.com/nuxeo/nuxeo/pull/4168/files#diff-172201976b34d3f53d4238f8bd458f38R208
I assumed an empty password did not need to be replaced, but indeed, maybe better to make it a complete mystery (?)

Reply:
Sorry but the anchor doesn't seem to get me to the right place ?
About misteries, I think putting stars doesn't give away the password is empty

Reply:
Anchor seems broken because it's a large diff https://github.com/nuxeo/nuxeo/pull/4168/commits/ee3f00ccd1519b8dc0041c65f9fe91fabd322a17#diff-172201976b34d3f53d4238f8bd458f38R215

Anyway i'll do the change